Gems/Atom/RPI/Code/Source/RPI.Public/WindowContext.cpp

```diff
@@ -68,27 +71,33 @@ namespace AZ
 
         const RHI::Ptr<RHI::SwapChain>& WindowContext::GetSwapChain(SwapChainMode swapChainMode) const
         {
-            return m_swapChains[static_cast<uint32_t>(swapChainMode)];
+            uint32_t swapChainIndex = static_cast<uint32_t>(swapChainMode);
+            AZ_Assert(swapChainIndex < GetSwapChainsSize(), "Swapchain with index %i does not exist", swapChainIndex);
+            return m_swapChainsData[swapChainIndex].m_swapChain;
         }
 
         uint32_t WindowContext::GetSwapChainsSize() const
         {
-            return aznumeric_cast<uint32_t>(m_swapChains.size());
+            return aznumeric_cast<uint32_t>(m_swapChainsData.size());
         }
 
         const RHI::Viewport& WindowContext::GetViewport(SwapChainMode swapChainMode) const
         {
-            return m_defaultViewports[static_cast<uint32_t>(swapChainMode)];     
+            uint32_t swapChainIndex = static_cast<uint32_t>(swapChainMode);
```

Comment:
Nit: Could do a slight consolidation with `return GetSwapchain(swapChainMode).m_viewport` here and below for the scissor if you want to not duplicate the assertion logic